spec/models/miq_report/generator_spec.rb

```diff
@@ -0,0 +1,83 @@
+describe MiqReport::Generator do
+  before do
+    EvmSpecHelper.local_miq_server
+    @user = FactoryGirl.create(:user_with_group)
+    @time_profile_all = FactoryGirl.create(:time_profile_with_rollup, :tz => "UTC")
+    @host1 = FactoryGirl.create(:host)
+  end
+
+  describe "#generate" do
+    context "Memory Utilization Trends report (daily)" do
+      before :each do
+        @miq_report_profile_all = FactoryGirl.create(
+          :miq_report,
+          :db              => "VimPerformanceTrend",
+          :order           => "Ascending",
+          :sortby          => ["resource_name"],
+          :time_profile_id => @time_profile_all.id,
+          :db_options      => {:limit_col    => "max_derived_memory_available",
+                               :trend_col    => "derived_memory_used",
+                               :rpt_type     => "trend",
+                               :start_offset => 604_800,
+                               :end_offset   => 0,
+                               :trend_db     => "HostPerformance",
+                               :interval     => "daily",
+                               :target_pcts  => [70, 80, 90]})
+      end
+
+      it "returns one row for each host" do
+        used_mem_up = [700, 600, 500, 400]
+        @host2 = FactoryGirl.create(:host)
+        create_rollup(@host1, @time_profile_all, used_mem_up)
+        create_rollup(@host2, @time_profile_all, used_mem_up)
+        @miq_report_profile_all.generate_table(:userid => @user.userid)
+        expect(@miq_report_profile_all.table.data.size).to eq(2)
+      end
+
+      it "calculates 'UP' trend" do
+        used_mem_up = [700, 600, 500, 400]
```

Comment:
It looks like this should be checking for trend down instead of up due to the direction of the values in the array. These cases may be easier to read if you reverse the arrays and change the timestamp calculation in `create_rollup` to add `1.day` instead of subtract.